qa/rpc-tests/mempool_size_limit.py

```diff
@@ -0,0 +1,605 @@
+            tx_sent += last_tx_size
+
+        print(f"Mempool almost full: {usage} out of {NODE0_LIMIT_B}")
+        self.assert_limits_enforced()
+
+        mark_logs("Sending low fee transction, expecting failure", self.nodes, DEBUG_MODE)
+        try:
+            tx = txs.pop(0)
+            tx_hex = tx['hex']
+            tx_input = tx['fee']
+            assert(tx_input <= mininput)
+            txid = self.nodes[0].decoderawtransaction(tx_hex)['txid']
+            self.nodes[0].sendrawtransaction(tx_hex, True)
+            assert(False)
+        except JSONRPCException as e:
+            print("Ok")
+
+        self.assert_limits_enforced()
+        mpool = self.nodes[0].getrawmempool()
+        assert(txid not in mpool)
+
+        mark_logs("Sending high fee transaction, expecting success", self.nodes, DEBUG_MODE)
+        try:
+            tx = txs.pop()
+            tx_hex = tx['hex']
+            tx_input = tx['fee']
+            assert(tx_input > mininput)
+            txid = self.nodes[0].decoderawtransaction(tx_hex)['txid']
+            self.nodes[0].sendrawtransaction(tx_hex, True)
+            print("Ok")
+        except JSONRPCException as e:
+            assert(False)
+
+        prev_mpool = mpool
+        mpool = self.nodes[0].getrawmempool()
+        assert(txid in mpool)
+        assert_equal(len(mpool), len(prev_mpool))
+
+        self.assert_limits_enforced()
+
+        print("Wait for high fee transaction to be present in node1 mempool")
+        timeout = 5
+        waiting = 0
+        mpool1 = self.nodes[1].getrawmempool()
+        while txid not in mpool1 and waiting < timeout:
+            print("Waiting...")
+            waiting += 1
+            time.sleep(1)
+            mpool1 = self.nodes[1].getrawmempool()
+
+        assert(txid in mpool1)
+        assert(len(mpool1) < len(mpool))
+        for tx1 in mpool1:
+            assert(tx1 in mpool)
+            txinfo = self.nodes[1].getrawmempool(True)[tx1]
+            feerate = txinfo['fee'] / txinfo['size']
+            assert(feerate >= minfeerate or tx1 in ctransactions)
+
+        # make sure that chained transactions, known to have aggregated high fee, have not been evicted
+        for ct in ctransactions:
+            assert(ct in mpool)
+            assert(ct in mpool1)
+
+        mark_logs("Sending low fee non ceasing certificates, expecting success", self.nodes, DEBUG_MODE)
+        cert_fees = []
+        nc_certs = [self.nodes[0].sendrawtransaction(certs[2][0]['hex']),
+                    self.nodes[0].sendrawtransaction(certs[3][0]['hex'])]
+        cert_sent_size = (len(certs[2][0]['hex']) + len(certs[3][0]['hex'])) // 2
+        cert_sent = 2
+        cert_fees.append(certs[2][0]['fee'] / (len(certs[2][0]['hex']) // 2))
+        cert_fees.append(certs[3][0]['fee'] / (len(certs[3][0]['hex']) // 2))
+
+        mark_logs("Filling mpool with low fee certificates, expecting success and transaction eviction", self.nodes, DEBUG_MODE)
```

Comment:
Similarly to previous comment, it could be worth checking that at least the lowest fee transaction is evicted (I know possibly the eviction affects multiple transactions, but the lowest fee for sure).

Reply:
Good point, but I believe that would require a new data set, or using the current data set in a different way. Currently we might have multiple transactions with the same minimum fee rate, and their ordering is not defined

Reply:
Agreed to handle this in a separate PR.